packages/live-share/src/schema-utils.ts

```diff
@@ -0,0 +1,117 @@
+        ([key, ObjectClass]) => {
+            return [key, getLiveDataObjectClassProxy(ObjectClass, liveRuntime, existingProxyRegistries)];
+        }
+    );
+    const newInitialObjects: LoadableObjectClassRecord =
+        Object.fromEntries(initialObjectEntries);
+
+    return {
+        initialObjects: newInitialObjects,
+        dynamicObjectTypes: schema.dynamicObjectTypes?.map((ObjectClass) =>
+            getLiveDataObjectClassProxy(ObjectClass, liveRuntime, existingProxyRegistries)
+        ),
+    };
+}
+
+/**
+ * @hidden
+ * Inject Live Share dependencies to relevant `LiveDataObject` derived classes.
+ * Regular `DataObject` classes are not proxied.
+ */
+export function getLiveDataObjectClassProxy<TClass extends IFluidLoadable>(
+    ObjectClass: LoadableObjectClass<any>,
+    liveRuntime: LiveShareRuntime,
+    existingProxyRegistries: Map<string, LoadableObjectClass<any>> = new Map(),
+): LoadableObjectClass<TClass> {
+    if (isLiveDataObject(ObjectClass)) {
+        // We should only be proxying one Live Share DDS per type.
+        // This is because Fluid attempts to de-duplicate by comparing classes, but we are dynamically creating proxies.
+        // They then enforce this de-duplication using the factory type name, throwing an error in `parseDataObjectsFromSharedObjects`.
+        // So, we ensure that we only create the proxy once per container.
+        const typeName = (ObjectClass as any).TypeName;
+        const CheckExisting = existingProxyRegistries.get(typeName);
+        if (CheckExisting !== undefined) {
+            return CheckExisting;
+        }
+        // Create a new proxy for this type and insert it into proxiedClasses
+        const NewProxy = proxyLiveDataObjectClass(
+            ObjectClass,
+            liveRuntime
+        ) as unknown as LoadableObjectClass<TClass>;
+        existingProxyRegistries.set(typeName, NewProxy);
+        return NewProxy;
+    }
+    return ObjectClass;
+}
+
+/**
+ * @hidden
+ */
+function isLiveDataObject(value: any): value is typeof LiveDataObject {
+    return value.LiveEnabled === true;
+}
+
+/**
+ * @hidden
+ * Create a new class extending LiveDataObject to inject in _liveRuntime
+ */
+function proxyLiveDataObjectClass<I extends DataObjectTypes = DataObjectTypes>(
+    BaseClass: typeof LiveDataObject<I>,
+    runtime: LiveShareRuntime
+): typeof LiveDataObject<I> {
+    class ProxiedBaseClass extends (BaseClass as unknown as new (
+        props: IDataObjectProps<I>
+    ) => LiveDataObject<I>) {
+        constructor(props: IDataObjectProps<I>) {
+            super(props);
+            this["liveRuntime"] = runtime;
+        }
+    }
+
+    const DynamicClass: typeof LiveDataObject<I> = class extends BaseClass {
+        public static TypeName = (BaseClass as any).TypeName;
+        public static readonly factory = new Proxy((BaseClass as any).factory, {
+            get: function (target, prop, receiver) {
+                if (prop === "ctor") {
```

Comment:
ctor is one of the params for the `factory` method we are reflecting. it stands for "constructor", since you are passing them the constructor for the class they are building. We are returning ProxiedBaseClass, but this is duck typing for a class that doesn't really have a type since it is created on the fly. If you can get it to compile with anything cleaner than this though let me know...this is as clean as I could figure out. It's far from perfect